parla/parray/memory.py

```diff
@@ -362,8 +362,24 @@ def get_by_global_slices(self, device_id: int, global_slices: SlicesType):
             global_slices: slice/ints/tuple/list<int>, use the same format as advance indexing of numpy
 
         Return
-            :class:`cupy.ndarray` or :class:`numpy.array` object
-        """
+            :class:`cupy.ndarray` or :class:`numpy.array` object or `None` if there is no copy at that device
+        """
+        # check if a copy exists at this device
+        #
+        # This is needed for the usage of @spawn() annotation
+        # sometimes device has no copy but still need a parray slices view object in advance
+        # Example:
+        # A = parray(numpy_array)
+        # @spawn(inout=[A], placement=gpu(0))
+        # def task1():
+        #   ... a task move A to GPU. free CPU copy ...
+        # @spawn(inout=[A[0]], placement=gpu(1))
```

Comment:
then A[0] is from gpu0? (IIUC, this originally was from CPU and since the CPU instance was freed it should be from GPU. Is it true? or am I missing?)

Reply:
Yes you are right